### DISCUSSION TITLE
`Import Page...` should detect Zim pages

### DISCUSSION THREAD

**Discussion by bruvellu** [zim-desktop-wiki/zim-desktop-wiki#2508] [Development]
Hello, I want to import pages from one Zim notebook from another. But if I import a Zim page using `File` > `Import Page...`, a new header is prepended to the file's content by default (and the original header becomes content):

![importpage](https://github.com/zim-desktop-wiki/zim-desktop-wiki/assets/361179/1dd2e0e5-f26d-49b7-a035-8aa898ad9ec5)

I saw there's a TODO on how to detect file formats in the [`ImportPageDialog`](https://github.com/zim-desktop-wiki/zim-desktop-wiki/blob/c88cf3cb53896bf272e87704826b77e82eddb3ef/zim/gui/uiactions.py#L531):

https://github.com/zim-desktop-wiki/zim-desktop-wiki/blob/c88cf3cb53896bf272e87704826b77e82eddb3ef/zim/gui/uiactions.py#L531

I think Zim should at least detect the Zim's format ;) My question is: Where should this logic be, in the `ImportPageDialog` or in the [`import_file`](https://github.com/zim-desktop-wiki/zim-desktop-wiki/blob/c88cf3cb53896bf272e87704826b77e82eddb3ef/zim/import_files.py#L13) function? And would checking for `Content-Type: text/x-zim-wiki` in the first line be a sufficiently robust method to detect a Zim filetype?

PS: I know I can just move/copy the file itself using the file browser, but I'd prefer to do it within Zim without opening other applications.

**Comment by introt** [comment]
> Where should this logic be, in the ImportPageDialog or in the [import_file](https://github.com/zim-desktop-wiki/zim-desktop-wiki/blob/c88cf3cb53896bf272e87704826b77e82eddb3ef/zim/import_files.py#L13) function? And would checking for Content-Type: text/x-zim-wiki in the first line be a sufficiently robust method to detect a Zim filetype?

@jaap-karssenberg
